src/commands/new.ts

```diff
@@ -12,84 +12,291 @@ import {
     }
+    // #endregion
+
+    // #region Bundle Identifier
+    const defaultBundleIdentifier = `com.${projectName.toLowerCase()}`
+    let bundleIdentifier = useDefault(options.bundle) ? defaultBundleIdentifier : options.bundle
+
+    if (bundleIdentifier === undefined) {
+      const bundleIdentifierResponse = await prompt.ask({
+        type: "input",
+        name: "bundleIdentifier",
+        message: "What bundle identifier?",
+        initial: defaultBundleIdentifier,
+      })
+
+      bundleIdentifier = bundleIdentifierResponse.bundleIdentifier
+    }
+
+    const { validateBundleIdentifier } = require("../tools/validations") as ValidationsExports
+    validateBundleIdentifier(toolbox, bundleIdentifier)
+
+    // #endregion
 
-    // custom bundle identifier (android only)
-    // TODO: refactor alert, need to rethink this
-    const bundleIdentifier =
-      validateBundleIdentifier(toolbox, parameters.options.bundle) ||
-      `com.${projectName.toLowerCase()}`
+    // #region Project Path
+    const defaultTargetPath = path(projectName)
+    let targetPath = useDefault(options.targetPath) ? defaultTargetPath : options.targetPath
+    if (targetPath === undefined) {
+      const targetPathResponse = await prompt.ask({
+        type: "input",
+        name: "targetPath",
+        message: "Where do you want to start your project?",
+        initial: defaultTargetPath,
+      })
+
+      targetPath = targetPathResponse.targetPath
+    }
+
+    // #endregion
+
+    // #region Overwrite
+    // if they pass in --overwrite, remove existing directory otherwise throw if exists
+    const defaultOverwrite = false
+    let overwrite = useDefault(options.overwrite) ? defaultOverwrite : options.overwrite
+    if (exists(targetPath)) {
+      if (overwrite === undefined) {
+        overwrite = await prompt.confirm(
+          `${targetPath} already exists. Do you want to overwrite it?`,
+          false,
+        )
+      }
+
+      if (overwrite === false) {
+        const alreadyExists = `Error: There's already a folder at ${targetPath}. To force overwriting that folder, run with --overwrite or say yes.`
+        p()
+        p(yellow(alreadyExists))
+        process.exit(1)
+      }
+
+      log(`Removing existing project ${projectName}`)
+      remove(projectName)
+    }
+    // #endregion
+
+    // #region Prompt Git Option
+    const defaultGit = true
+    let git = useDefault(options.git) ? defaultGit : options.git
+
+    if (git === undefined) {
+      git = await prompt.confirm("Do you want to initialize a git repository?", true)
+    }
+    // #endregion
 
+    // #region Packager
     // check if a packager is provided, or detect one
     // we pass in expo because we can't use pnpm if we're using expo
-    const packagerName = parameters.options.packager || packager.detectPackager()
+
+    const availablePackagers = packager.availablePackagers()
+    const defaultPackagerName = availablePackagers.includes("yarn") ? "yarn" : "npm"
```

Comment:
According to this twitter poll, it seems that people overwhelming prefer `yarn` for React Native development: https://twitter.com/JoshuaYoes/status/1557809021905563649

Previously, we preferred pnpm -> yarn -> npm. Anecdotally, I had a lot of issues with using `pnpm`, so I think it would be best to prefer yarn in this case

Reply:
That makes sense, and I think my enthusiasm for pnpm has been tempered since I made this change.

Reply:
Honestly, pnpm has been an amazing package manager for me for traditional JS projects. Install times are pretty unbeatable. But I think their strategy of system links seems to be more buggy with native dependencies for react native